src/app/context/CurrencyContext.tsx

```diff
@@ -0,0 +1,310 @@
+}
+
+const CurrencyContext = createContext({} as CurrencyContextType);
+
+const numSatsInBtc = 100_000_000;
+
+export function CurrencyProvider({ children }: { children: React.ReactNode }) {
+  const currencies = [
+    "AED",
+    "AFN",
+    "ALL",
+    "AMD",
+    "ANG",
+    "AOA",
+    "ARS",
+    "AUD",
+    "AWG",
+    "AZN",
+    "BAM",
+    "BBD",
+    "BDT",
+    "BGN",
+    "BHD",
+    "BIF",
+    "BMD",
+    "BND",
+    "BOB",
+    "BRL",
+    "BSD",
+    "BTC",
+    "BTN",
+    "BWP",
+    "BYR",
+    "BZD",
+    "CAD",
+    "CDF",
+    "CHF",
+    "CLF",
+    "CLP",
+    "CNY",
+    "COP",
+    "CRC",
+    "CUP",
+    "CVE",
+    "CZK",
+    "DJF",
+    "DKK",
+    "DOP",
+    "DZD",
+    "EEK",
+    "EGP",
+    "ERN",
+    "ETB",
+    "EUR",
+    "FJD",
+    "FKP",
+    "GBP",
+    "GEL",
+    "GHS",
+    "GIP",
+    "GMD",
+    "GNF",
+    "GTQ",
+    "GYD",
+    "HKD",
+    "HNL",
+    "HRK",
+    "HTG",
+    "HUF",
+    "IDR",
+    "ILS",
+    "INR",
+    "IQD",
+    "IRR",
+    "ISK",
+    "JEP",
+    "JMD",
+    "JOD",
+    "JPY",
+    "KES",
+    "KGS",
+    "KHR",
+    "KMF",
+    "KPW",
+    "KRW",
+    "KWD",
+    "KYD",
+    "KZT",
+    "LAK",
+    "LBP",
+    "LKR",
+    "LRD",
+    "LSL",
+    "LTL",
+    "LVL",
+    "LYD",
+    "MAD",
+    "MDL",
+    "MGA",
+    "MKD",
+    "MMK",
+    "MNT",
+    "MOP",
+    "MRO",
+    "MTL",
+    "MUR",
+    "MVR",
+    "MWK",
+    "MXN",
+    "MYR",
+    "MZN",
+    "NAD",
+    "NGN",
+    "NIO",
+    "NOK",
+    "NPR",
+    "NZD",
+    "OMR",
+    "PAB",
+    "PEN",
+    "PGK",
+    "PHP",
+    "PKR",
+    "PLN",
+    "PYG",
+    "QAR",
+    "RON",
+    "RSD",
+    "RUB",
+    "RWF",
+    "SAR",
+    "SBD",
+    "SCR",
+    "SDG",
+    "SEK",
+    "SGD",
+    "SHP",
+    "SLL",
+    "SOS",
+    "SRD",
+    "STD",
+    "SVC",
+    "SYP",
+    "SZL",
+    "THB",
+    "TJS",
+    "TMT",
+    "TND",
+    "TOP",
+    "TRY",
+    "TTD",
+    "TWD",
+    "TZS",
+    "UAH",
+    "UGX",
+    "USD",
+    "UYU",
+    "UZS",
+    "VEF",
+    "VND",
+    "VUV",
+    "WST",
+    "XAF",
+    "XAG",
+    "XAU",
+    "XBT",
+    "XCD",
+    "XDR",
+    "XOF",
+    "XPF",
+    "YER",
+    "ZAR",
+    "ZMK",
+    "ZMW",
+    "ZWL",
+  ];
+
+  const [balances, setBalances] = useState<{
+    satsBalance: string;
+    fiatBalance: string;
+  }>({ satsBalance: "", fiatBalance: "" });
+  const [currency, setCurrency] = useState<SupportedCurrencies>("USD");
+  const [exchange, setExchange] = useState<SupportedExchanges>("Coindesk");
+  const auth = useAuth();
+
+  const bitcoinToFiat = async (
+    amountInBtc: number | string,
+    convertTo: SupportedCurrencies
+  ) => {
+    const btc = new Decimal(amountInBtc);
+    const rate = await getFiatBtcRate(convertTo);
+    return btc.mul(rate).toNumber();
+  };
+
+  const getFiatBtcRate = async (
+    currency: SupportedCurrencies
+  ): Promise<string> => {
+    let response;
+
+    if (exchange === "Yad.io") {
```

Comment:
I guess this is fine for now. If we later add support for more APIs we should create something which handles different API-responses in a different place maybe.

Reply:
Yea open to supporting this and change it in the future if a ticket comes into support a specific exchange.